source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -328,15 +340,10 @@ namespace PlayFab
     {
         try
         {
-            // batch wasn't sent out due to an error, find it in the batch tracking map using customData pointer as a key
-            const auto foundBatchIterator = this->batchesInFlight.find(customData);
-            if (foundBatchIterator == this->batchesInFlight.end())
-            {
-                LOG_PIPELINE("Untracked batch was returned to EventsAPI.WriteEvents callback"); // normally this never happens
-            }
-            else
+            std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>> batchWritten;
+            if(TryGetBatchOutOfFlight(customData, &batchWritten))
             {
-                auto requestBatchPtr = std::shared_ptr<const std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>>(new std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>(std::move(foundBatchIterator->second)));
+                auto requestBatchPtr = std::shared_ptr<const std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>>(new std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>(batchWritten));
```

Comment:
you lost std::move when declaring these requestBatchPtr values. I think you still want to keep those or else you will copy the entire batchWritten vector unnecessarily

Reply:
I thought I was moving the std::move to the batchReturn. But it still makes sense to be moving the object if it is movable.

Reply:
you are correct, you are std::move-ing into batchReturn
but batchWritten still needs to be std::moved if you want to avoid the copy when you construct the new shared_ptr variable. Shared_ptr is an owning pointer which means it won't take a weak reference to batchReturn. It will copy unless you use std::move

there shouldn't be a need to clear out `batchWritten` after you've std::moved it though